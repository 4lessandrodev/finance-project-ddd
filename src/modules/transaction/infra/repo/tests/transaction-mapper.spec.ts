import TransactionMock from "@modules/transaction/domain/tests/mock/transaction.mock";
import TransactionCalculationToDomain from "../transaction-calculation.mapper";
import TransactionToDomainMapper from "../transaction.mapper";

describe('transaction.mapper', () => {

	const mock = new TransactionMock();
	const calcMapper = new TransactionCalculationToDomain();
	const mapper = new TransactionToDomainMapper(calcMapper);

	it('should translate to domain with success', () => {
		const model = mock.model();
		const result = mapper.map(model);
		const generated = result.getResult().toObject();
		
		expect(result.isSuccess).toBeTruthy();
		expect(generated).toEqual(model);
		expect(generated).toMatchSnapshot();
		
	});
});
