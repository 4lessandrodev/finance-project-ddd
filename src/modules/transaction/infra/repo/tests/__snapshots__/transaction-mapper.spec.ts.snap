// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transaction.mapper should translate to domain with success 1`] = `
Object {
  "attachment": "http://s3-us-east-1.amazonaws.com/bucket/file.pdf",
  "createdAt": 2022-01-01T00:00:00.000Z,
  "deletedAt": undefined,
  "id": "valid_id",
  "isDeleted": false,
  "note": "valid_note",
  "paymentDate": 2022-01-01T00:00:00.000Z,
  "reason": "valid_reason_id",
  "status": "CONCLUIDO",
  "totalValue": Object {
    "currency": "BRL",
    "value": 1000,
  },
  "transactionCalculations": Array [
    Object {
      "budgetBoxId": "valid_budget_box_id",
      "budgetBoxName": "valid_name",
      "currency": Object {
        "currency": "BRL",
        "value": 1000,
      },
    },
  ],
  "transactionType": "ENTRADA",
  "updatedAt": 2022-01-01T00:00:00.000Z,
  "userId": "valid_user_id",
}
`;
